src/lib/efthrm/tcp_helper_cluster.c

```diff
@@ -1504,6 +1519,7 @@ int efab_tcp_helper_reuseport_bind(ci_private_t *priv, void *arg)
       drop_flags = 0;
     rc = thc_alloc_thr(thc, trb->cluster_restart_opt,
                        trb->cluster_hot_restart_opt,
+                       /* pktbuf_memfd */ -1,
                        &ni->opts,
                        ni->flags | drop_flags,
                        &thr);
```

Comment:
I read it as "Let's break nginx profile", because nginx works via this part of code and specifies `EF_USE_HUGE_PAGES` to 2.

Reply:
Thanks for pointing that out! It does look like a regression in Linux 5.7+. I'm looking at how to donate a memfd file descriptor to this workflow. I didn't do it in one batch with other suggestions, including internal, to reduce the number of changes in flight. Thanks again.

Reply:
Actually, it's probably a false alarm.

I remember fixing it specifically for `nginx` or `wrk` by introducing the allocator's reference counting for this and other clustering workflows:
```
-> efab_tcp_helper_reuseport_bind()
-> thc_alloc_thr()
-> tcp_helper_rm_alloc()
```
Where we bump a reference if the allocator is available:
```
  if( thc && thc->thc_pktbuf_alloc ) {
    rs->thc_pktbuf_alloc = oo_hugetlb_allocator_get(thc->thc_pktbuf_alloc);
```
The main problem was the undefined order of who is created first (`tcp_helper_cluster_t` or `tcp_helper_resource_t`) and then making sure that all `tcp_helper_resource_t` instances share the same allocator since they're the same cluster.

If there was a missing memfd donation, I would fix it by plumbing through `in_pktbuf_memfd`.

Let me add a comment in the code clarifying that: it's not obvious and I've already forgotten things enough so they look like a regression.